### DISCUSSION TITLE
num_retries variable for --write-out format

### DISCUSSION THREAD

**Discussion by JayGuerette** [curl/curl#13901] [Ideas]
It would be nice to expose the number of --retry retries that were required to complete a request:

`curl --retry 5 --retry-delay 2 --silent --write-out '%{num_retries} attempts\n' https://unreliable.api/wth/`

**Comment by bagder** [comment]
A great idea I think!

**Reply by bagder** [reply]
What should the counter say if you do multiple transfers? Accumulated or just for the last transfer? Something else?

Example:
~~~
curl --retry 5 --retry-delay 2 --silent --write-out '%{num_retries} attempts\n' \
https://unreliable.api/ https://unreliable.api/

**Reply by icing** [reply]
Isn't it a property of the transfer individually?

**Reply by bagder** [reply]
Hm, for some reason for a second there I was thinking the output would only be shown once for two transfers. Brain lapse. Never mind me, it should of course be per individual transfer.

**Comment by bagder** [comment]
See https://github.com/curl/curl/pull/13910

**Reply by JayGuerette** [reply]
Thanks!

**Reply by bagder** [reply]
Will ship in 8.9.0!
